docs/administration/Query_planning.md

```diff
@@ -193,6 +193,104 @@ By removing the specific expressions (only keep the operators), the query plan c
 
 ![8-5](../assets/8-5.png)
 
+## Query hint
+
+Query hints are directives or comments that explicitly suggest the query optimizer on how to execute a query. Hints only take effect within a single query.
+
+### System variable hint
+
+You can set one or more system variable hints by using the `/*+ ... */` comment syntax in SELECT and SUBMIT TASK statements, or in SELECT clauses that are included in other statements such as CREATE MATERIALIZED VIEW AS SELECT and CREATE VIEW AS SELECT.
+
+#### Syntax
+
+~~~ SQL
+[...] SELECT [/*+ SET_VAR(key=value [, key = value]*) */] ...
+SUBMIT [/*+ SET_VAR(key=value [, key = value]*) */] TASK ...
+~~~
+
+#### Examples
+
+ Hint the aggregation method for an aggregate query by setting the system variables `streaming_preaggregation_mode` and `new_planner_agg_stage`.
+
+~~~ SQL
+SELECT /*+ SET_VAR (streaming_preaggregation_mode = 'force_streaming',new_planner_agg_stage = '2') */ SUM(sales_amount) AS total_sales_amount FROM sales_orders;
+~~~
+
+Hint the query's task execution timeout period by setting the system variable `query_timeout` in the SUBMIT TASK statement.
+
+~~~ SQL
+SUBMIT /*+ SET_VAR(query_timeout=3) */ TASK AS CREATE TABLE temp AS SELECT count(*) AS cnt FROM tbl1;
+~~~
+
+Hint the query's execution timeout period by setting the system variable `query_timeout` in the SELECT clause when creating a materialized view.
+
+~~~ SQL
+CREATE MATERIALIZED VIEW mv 
+PARTITION BY dt 
+DISTRIBUTED BY HASH(`key`) 
+BUCKETS 10 
+REFRESH ASYNC 
+AS SELECT /*+ SET_VAR(query_timeout=500) */ * from dual;
+~~~
+
+### Join hint
+
+For multi-table join queries, the optimizer usually selects the optimal join execution method. In special cases, you can use a join hint to explicitly suggest the join execution method to the optimizer or disable Join Reorder. Currently, a join hint supports suggesting Shuffle Join, Broadcast Join, Bucket Shuffle Join, or Colocate Join as a join execution method. When a join hint is used, the optimizer does not perform Join Reorder. So you need to select the smaller table as the right table. Additionally, when suggesting Colocate Join or Bucket Shuffle Join as the join execution method, make sure that the data distribution of the joined table meets the requirements of these join execution methods. Otherwise, the suggested join execution method cannot take effect.
+
+#### Syntax
+
+~~~SQL
+... JOIN { [BROADCAST] | [SHUFFLE] | [BUCKET] | [COLOCATE] | [UNREORDER]} ...
+~~~
+
+> **NOTE**
+>
+> Join Hint is case-insensitive.
+
+#### Examples
+
+- Shuffle Join
+
+  If you need to shuffle the data rows with the same bucket key values from tables A and B onto the same machine before a Join operation is performed, you can hint the join execution method as Shuffle Join.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/03bb6664-ef3b-488c-a94b-537fb7f01907)
 should be "bucketing key" I think. There are two sentences with "bucket key" in this article.